### PR TITLE
SubGhz: add GFSK modulation, refactoring subghz_chat

### PR DESCRIPTION
**Pull request by Skorpionm**
# What's new

- SubGhz: subghz_tx_rx_worker, add GFSK modulation
- SubGhz: subghz_chat, refactoring
- SubGhz: subghz_chat, add UTF-8 support 

# Verification 

- Compile and download
- Take more flippers, download subghz_chat and pee in different languages

# Checklist (do not modify)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [X] Description contains actions to verify feature/bugfix
- [X] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/subghz/helpers/subghz_chat.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            event.c = c;
+            osMessageQueuePut(instance->event_queue, &event, 0, 0);
+        }
+    }
+
+    FURI_LOG_I(TAG, "Worker stop");
+    return 0;
+}
+
+static void subghz_chat_worker_udate_event_chat(void* context) {
```
Comment (anna-oake) [inline applications/subghz/helpers/subghz_chat.c]: update

Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.c]: +

File: applications/subghz/helpers/subghz_chat.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    SubghzChatEventNoEvent,
+    SubghzChatEventUserEntrance,
+    SubghzChatEventUserExit,
+    SubghzChatEventInputData,
+    SubghzChatEventRXData,
+    SubghzChatEventNewMessage,
+} SubghzChatEvent;
+
+typedef struct {
+    SubghzChatEvent event;
```
Comment (DrZlo13) [inline applications/subghz/helpers/subghz_chat.h]: SubghzChatEvent -> SubghzChatEventType

File: applications/subghz/helpers/subghz_chat.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    char c;
+} SubghzChatQueue;
```
Comment (DrZlo13) [inline applications/subghz/helpers/subghz_chat.h]: SubghzChatQueue -> SubghzChatEvent

File: applications/subghz/helpers/subghz_chat.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    SubghzChatQueue event;
+    if(osMessageQueueGet(instance->event_queue, &event, NULL, osWaitForever) == osOK) {
+        return event;
+    } else {
+        event.event = SubghzChatEventNoEvent;
+        return event;
+    }
+}
+
+void subghz_chat_worker_set_event_chat(SubGhzChatWorker* instance, SubghzChatQueue* event) {
```
Comment (DrZlo13) [inline applications/subghz/helpers/subghz_chat.c]: subghz_chat_worker_**put**_event_chat

File: applications/subghz/helpers/subghz_chat.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    furi_thread_set_stack_size(instance->thread, 2048);
+    furi_thread_set_context(instance->thread, instance);
+    furi_thread_set_callback(instance->thread, subghz_chat_worker_thread);
+    instance->subghz_txrx = subghz_tx_rx_worker_alloc();
+    instance->event_queue = osMessageQueueNew(80, sizeof(SubghzChatQueue), NULL);
+    return instance;
+}
+
+void subghz_chat_worker_free(SubGhzChatWorker* instance) {
+    furi_assert(instance);
```
Comment (DrZlo13) [inline applications/subghz/helpers/subghz_chat.c:68]: furi_assert(!instance->worker_running)

File: applications/subghz/helpers/subghz_chat.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+SubGhzChatWorker* subghz_chat_worker_alloc();
+void subghz_chat_worker_free(SubGhzChatWorker* instance);
+bool subghz_chat_worker_start(SubGhzChatWorker* instance, uint32_t frequency);
+void subghz_chat_worker_stop(SubGhzChatWorker* instance);
+bool subghz_chat_worker_is_running(SubGhzChatWorker* instance);
+SubghzChatQueue subghz_chat_worker_get_event_chat(SubGhzChatWorker* instance);
+void subghz_chat_worker_set_event_chat(SubGhzChatWorker* instance, SubghzChatQueue* event);
```
Comment (DrZlo13) [inline applications/subghz/helpers/subghz_chat.h]: subghz_chat_worker_**put**_event_chat

File: lib/subghz/subghz_tx_rx_worker.c [hunk 7]
```diff
@@ -108,15 +110,13 @@ bool subghz_tx_rx_worker_rx(SubGhzTxRxWorker* instance, uint8_t* data, uint8_t*
...
+    uint8_t timeout = 200;
     if(instance->satus != SubGhzTxRxWorkerStatusIDLE) {
         furi_hal_subghz_idle();
     }
     furi_hal_subghz_write_packet(data, size);
-    instance->satus = SubGhzTxRxWorkerStatusTx;
-
     furi_hal_subghz_tx(); //start send
-
+    instance->satus = SubGhzTxRxWorkerStatusTx;
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: instance->s**t**atus

**Review by DrZlo13 — CHANGES_REQUESTED** [review]

File: lib/subghz/subghz_tx_rx_worker.c (hunk 7 above)
Comment (Skorpionm) [reply lib/subghz/subghz_tx_rx_worker.c]: Ok

File: applications/subghz/helpers/subghz_chat.h (hunk 6 above)
Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.h]: Ok

File: applications/subghz/helpers/subghz_chat.c (hunk 5 above)
Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.c:68]: Ok

File: applications/subghz/helpers/subghz_chat.c (hunk 4 above)
Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.c]: Ok

File: applications/subghz/helpers/subghz_chat.h (hunk 3 above)
Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.h]: Ok

File: applications/subghz/helpers/subghz_chat.h (hunk 2 above)
Comment (Skorpionm) [reply applications/subghz/helpers/subghz_chat.h]: Ok

**Review by skotopes — APPROVED** [review]

**Review by DrZlo13 — APPROVED** [review]

**Pull request merged**